test/e2e/metrics_test.go

```diff
@@ -156,11 +158,21 @@ func (c *MetricsTestConfig) validate(token string) {
 	require.Contains(c.t, string(output), "200 OK", "Metrics endpoint did not return 200 OK")
 }
 
-// cleanup created resources
+// cleanup removes the created resources. Uses a context with timeout to prevent hangs.
 func (c *MetricsTestConfig) cleanup() {
 	c.t.Log("Cleaning up resources")
-	_ = exec.Command(c.client, "delete", "clusterrolebinding", c.clusterBinding, "--ignore-not-found=true").Run()
-	_ = exec.Command(c.client, "delete", "pod", c.curlPodName, "-n", c.namespace, "--ignore-not-found=true").Run()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
This is interesting. What about adding a check to see if the clusterbinding and pod is actually deleted and then move forward.

Reply:
yeah that might be more direct fix. I'll try it after this one shows 10/10 pass rate locally.

Reply:
I added the checks in the latest commit, could resolve faster this way.